jaxrs/src/main/java/com/cerner/beadledom/jaxrs/exceptionmappers/FailureExceptionMapper.java

```diff
@@ -0,0 +1,70 @@
+package com.cerner.beadledom.jaxrs.exceptionmappers;
+
+import static javax.ws.rs.core.Response.Status.INTERNAL_SERVER_ERROR;
+
+import com.cerner.beadledom.jaxrs.models.JsonError;
+
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.ext.ExceptionMapper;
+import javax.ws.rs.ext.Provider;
+
+import org.jboss.resteasy.spi.Failure;
+import org.jboss.resteasy.spi.ReaderException;
+import org.jboss.resteasy.spi.WriterException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Exception Mapper for all unhandled Exceptions extending from the {@link Failure} hierarchy
+ * (specifically for Resteasy Built-in Internally-Thrown Exceptions like {@link ReaderException},
+ * {@link WriterException}, etc.) thrown by the service. See
+ * http://docs.jboss.org/resteasy/docs/3.1.4.Final/userguide/html_single/index.html#builtinException
+ * for more information.
+ *
+ * @author Cal Fisher
+ */
+@Provider
+public class FailureExceptionMapper implements ExceptionMapper<Failure> {
+
+  private static final Logger logger = LoggerFactory.getLogger(FailureExceptionMapper.class);
+
+  /**
+   * Generates a Response object with a Status of the Failure, Content-Type 'application/json', and
+   * a JsonError entity containing details about the unhandled exception in Json format.
+   *
+   * @param exception the Failure exception that was not handled
+   * @return a json response with the exception's status or 500 if the exception has no response
+   */
+  @Override
+  public Response toResponse(Failure exception) {
+
+    logger.error("An unhandled Failure exception was thrown.", exception);
+
+    int statusCode;
+    if (exception.getResponse() != null) {
+      statusCode = exception.getResponse().getStatus();
+    } else {
+      statusCode = INTERNAL_SERVER_ERROR.getStatusCode();
+    }
```

Comment:
if we wanted to keep this logic could we condense it a bit?

```java
Response response = exception.getResponse();

int statusCode = response == null ? INTERNAL_SERVER_ERROR.getStatusCode() : response.getStatusCode();
```

or maybe even

```java
int code = e.getResponse() == null ? INTERNAL_SERVER_ERROR.getStatusCode() : e.getResponse().getStatus();
```

I'm not sure what will fit best on a line.

Reply:
Could also go multiline:

```java
int code = e.getResponse() == null 
    ? INTERNAL_SERVER_ERROR.getStatusCode() 
    : e.getResponse().getStatus();
```

Reply:
https://github.com/cerner/beadledom/pull/45/commits/ec2febdf850acc1686b13715c230407d77733959#diff-94f0fc074c60a4a4d74f16bebf75afc9R44